### PR TITLE
[16.0][MIG] website_sale_vat_required: Migration to version 16.0

### PR DESCRIPTION
**Pull request by pilarvargas-tecnativa**
Superseed: https://github.com/OCA/e-commerce/pull/783

Pending from: 
- [x] https://github.com/OCA/e-commerce/pull/843  - Fix test errors in website_sale_checkout_skip_payment module 

This PR includes the modifications made in the migration of the other PR and the modification of the tests so that we do not have to depend on the payment methods module to run the tests of the functionality of this module.

cc @Tecnativa TT44388

@chienandalu @CarlosRoca13 please review :)

### CODE REVIEW

File: website_sale_vat_required/tests/test_website_sale_vat_required.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+@odoo.tests.tagged("post_install", "-at_install")
+class TestWebsiteSaleVatRequired(HttpCase):
+    def test_website_sale_vat_required(self):
+        self.browser_js(
+            url_path="/",
+            code="odoo.__DEBUG__.services['web_tour.tour'].run('shop_buy_product')",
+            ready="odoo.__DEBUG__.services['web_tour.tour']"
+            ".tours.shop_buy_product.ready",
+            login=None,
+        )
```
Comment (chienandalu) [inline website_sale_vat_required/tests/test_website_sale_vat_required.py]: Use the modern call (`start_tour`)

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: website_sale_vat_required/tests/test_website_sale_vat_required.py (hunk 1 above)
Comment (pilarvargas-tecnativa) [reply website_sale_vat_required/tests/test_website_sale_vat_required.py]: thanks :)

**Review by Abranes — APPROVED** [review]
LGTM

Code review

File: website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js [hunk 2]
```diff
@@ -9,22 +9,38 @@ odoo.define("website_sale_vat_required.tour", function (require) {
 
     steps.splice(
         _.findIndex(steps, {content: "go to checkout"}) + 1,
-        0,
+        4,
```
Comment (chienandalu) [inline website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js]: This makes no sense to me and could cause issues with other modules. You are taking this chunk of the orignial test away: https://github.com/odoo/odoo/blob/a0c1e2aa602ae46598a350ea6ae8d8b4a0c1c823/addons/website_sale/static/tests/tours/website_sale_buy.js#L63-L87

I'm thinking that `website_sale_checkout_skip_payment` should `splice` its part so we don't have to be aware of it on each tour that uses this

Comment (pilarvargas-tecnativa) [reply website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js]: This test already has its own steps :)

Comment (chienandalu) [reply website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js]: Rerun Odoo tests, maybe there was a temporary glitch making them fail on a tour

**Review by chienandalu — APPROVED** [review]

File: website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js (hunk 2 above)
Comment (pilarvargas-tecnativa) [reply website_sale_vat_required/static/src/js/website_sale_vat_required.tour.js]: It's always the same...

**Comment by pilarvargas-tecnativa** [comment]
@pedrobaeza  can this be merged or does @CarlosRoca13 need to take a look at it?

**Review by CarlosRoca13 — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@pilarvargas-tecnativa please rebase and check the problem.

**Comment by pilarvargas-tecnativa** [comment]
> @pilarvargas-tecnativa please rebase and check the problem.

solved!

**Pull request merged**